### DISCUSSION TITLE
[Twitter] Twitter → Nitter

### DISCUSSION THREAD

**Discussion by Palumgarus** [mshibanami/redirect-web#27] [Redirect Rules]
This redirect rule will take you from Twitter to the alternative privacy-friendly front-end Nitter (**provided you have a working instance**).

* Redirect from `^(https?:\/\/)(www\.)?(twitter\.com|t\.co|x\.com)\/?(@?[a-zA-Z0-9_]+\/)?(status\/\d+)?(\/[^\/?]+\/[a-zA-Z0-9_]+\/?)?([^?]*)(\?.*)?$`
* Redirect to `https://nitter.privacydev.net/$4$5$6$7`
 
 _Examples:_

* `https://twitter.com/` → `https://libreddit.privacydev.net/`
* `twitter.com` → `https://nitter.privacydev.net/`
* `https://x.com/jack/status/20` → `https://nitter.privacydev.net/jack/status/20`
* `https://twitter.com/rsa/status/55281?ref_src=twsrc%5Etfw%7Ctwcamp%5Etweetembed%7Ctwterm%5E55281%7Ctwgr%5E33f1c8f22fefb168a40e43cca5e870989ef366a9%7Ctwcon%5Es1_&ref_url=https%3A%2F%2Fqz.com%2Fembed%2Finset%2Fiframe%3Fid%3Dtwitter-55281autosize%3D1` → `https://nitter.privacydev.net/rsa/status/55281`

<details><summary>Export of the redirect rule</summary>
<p>

```
{
  "appVersion" : "5.1.1",
  "bundleID" : "io.github.mshibanami.RedirectWebForSafari",
  "createdAt" : "2024-04-26T20:16:53.514Z",
  "kind" : "RedirectList",
  "redirects" : [
    {
      "captureGroupProcesses" : [

      ],
      "comments" : "Breakdown:\n1. `^(https?:\\\/\\\/)?`: This part checks for the presence of `http:\/\/` or `https:\/\/` at the beginning of the URL. The `?` makes this part optional.\n2. `(www\\.)?`: This checks for the presence of `www.` at the beginning of the domain. The `?` makes this part optional.\n3. `(twitter\\.com|t\\.co|x\\.com)`: This checks for one of the specified domain names: `twitter.com`, `t.co`, or `x.com`.\n4. `\\\/?`: This checks for an optional slash `\/` after the domain.\n5. `(@?[a-zA-Z0-9_]+\\\/)?`: This checks for an optional Twitter username followed by a slash.\n6. `(status\\\/\\d+)?`: This checks for the pattern `status\/` followed by one or more digits.\n7. `(\\\/[^\\\/?]+\\\/[a-zA-Z0-9_]+\\\/?)?`: This checks for an optional path segment in the URL.\n8. `([^?]*)`: This captures any characters except `?` zero or more times, representing the remaining part of the URL without the query string.\n9. `(\\?.*)?`: This captures any query string if present, starting with `?`.\n\nThese examples should provide a good overview of the types of Twitter\/X URLs that the regex can accurately capture:\n1. https:\/\/twitter.com\/\n2. https:\/\/twitter.com\/username\n3. https:\/\/twitter.com\/username\/status\/1234567890\n4. https:\/\/twitter.com\/username\/segment\/segment2\n5. http:\/\/www.twitter.com\/\n6. http:\/\/www.twitter.com\/username\n7. http:\/\/www.twitter.com\/username\/status\/1234567890\n8. http:\/\/www.twitter.com\/username\/segment\/segment2\n9. https:\/\/t.co\/\n10. https:\/\/t.co\/abcdefg\n11. https:\/\/t.co\/abcdefg?utm_source=twitter\n12. https:\/\/x.com\/\n13. https:\/\/x.com\/username\n14. https:\/\/x.com\/username\/segment\/segment2\n15. http:\/\/www.x.com\/\n16. http:\/\/www.x.com\/username\n17. http:\/\/www.x.com\/username\/segment\/segment2",
      "destinationURLPattern" : "https:\/\/nitter.privacydev.net\/$4$5$6$7",
      "exampleURLs" : [
        "https:\/\/twitter.com",
        "twitter.com",
        "https:\/\/x.com\/jack\/status\/20",
        "https:\/\/twitter.com\/rsa\/status\/55281?ref_src=twsrc%5Etfw%7Ctwcamp%5Etweetembed%7Ctwterm%5E55281%7Ctwgr%5E33f1c8f22fefb168a40e43cca5e870989ef366a9%7Ctwcon%5Es1_&ref_url=https%3A%2F%2Fqz.com%2Fembed%2Finset%2Fiframe%3Fid%3Dtwitter-55281autosize%3D1"
      ],
      "excludeURLPatterns" : [

      ],
      "isEnabled" : true,
      "kind" : "Redirect",
      "sourceURLPattern" : {
        "type" : "regularExpression",
        "value" : "^(https?:\\\/\\\/)?(www\\.)?(twitter\\.com|t\\.co|x\\.com)\\\/?(@?[a-zA-Z0-9_]+\\\/)?(status\\\/\\d+)?(\\\/[^\\\/?]+\\\/[a-zA-Z0-9_]+\\\/?)?([^?]*)(\\?.*)?$"
      },
      "title" : "Twitter Front-End (Nitter)"
    }
  ]
}
```

</p>
</details>

To use an other instance just change `nitter.privacydev.net` in `"destinationURLPattern" : "https:\/\/nitter.privacydev.net\/$4$5$6$7"`.

<details><summary>How to use exports</summary>
<p>

1. Create a `.txt` document
2. Copy the export’s text
3. Paste it in the document
4. Change the document’s extension to `.redirectweb`
5. Use the share menu to open the document in Redirect Web

</p>
</details>

For convenience I’ve attached the redirect rule to the front-end, just change the extension from `.txt` to `.redirectweb` then use the share menu to open the document in Redirect Web.

[[Redirect Web] Twitter Front-End (Nitter).txt](https://github.com/mshibanami/redirect-web/files/15134981/Redirect.Web.Twitter.Front-End.Nitter.txt)

**Comment by iyerusad** [comment]
Is there a way to exclude redirecting when using the nitter button to view a page on twitter?
![image](https://github.com/mshibanami/redirect-web/assets/25647368/095690b1-b63f-46d4-9e6c-d709ede4c7b4)
